source/blood/src/nnexts.h

```diff
@@ -0,0 +1,271 @@
+kModernThingThrowableRock           = 434, // does small damage if hits target
+kModernThingEnemyLifeLeech          = 435, // the same as normal, except it aims in specified target only
+kModernPlayerControl                = 500, /// WIP
+kGenModernMissileUniversal          = 704,
+kGenModernSound                     = 708,
+};
+
+// type of random
+enum {
+kRandomizeItem                      = 0,
+kRandomizeDude                      = 1,
+kRandomizeTX                        = 2,
+};
+
+// type of object
+enum {
+OBJ_WALL                            = 0,
+OBJ_SPRITE                          = 3,
+OBJ_SECTOR                          = 6,
+};
+
+
+// - STRUCTS ------------------------------------------------------------------
+struct SPRITEMASS { // sprite mass info for getSpriteMassBySize();
+    int seqId;
+    short picnum; // mainly needs for moving debris
+    short xrepeat;
+    short yrepeat;
+    short clipdist; // mass multiplier
+    int mass;
+    short airVel; // mainly needs for moving debris
+    int fraction; // mainly needs for moving debris
+};
+
+struct QAVSCENE { // this one stores qavs anims that can be played by trigger
+    short index = -1;  // index of sprite which triggered qav scene
+    QAV* qavResrc = NULL;
+    short dummy = -1;
+};
+
+struct THINGINFO_EXTRA {
+    bool allowThrow; // indicates if kDudeModernCustom can throw it
+};
+
+struct VECTORINFO_EXTRA {
+    int fireSound[2]; // predefined fire sounds. used by kDudeModernCustom, but can be used for something else.
+};
+
+struct MISSILEINFO_EXTRA {
+    int fireSound[2]; // predefined fire sounds. used by kDudeModernCustom, but can be used for something else.
+    bool dmgType[kDamageMax]; // list of damages types missile can use
+};
+
+struct TRPLAYERCTRL { // this one for controlling the player using triggers (movement speed, jumps and other stuff)
+    QAVSCENE qavScene;
+};
+
+// - VARIABLES ------------------------------------------------------------------
+extern bool gModernMap;
+extern bool gTeamsSpawnUsed;
+extern ZONE gStartZoneTeam1[kMaxPlayers];
+extern ZONE gStartZoneTeam2[kMaxPlayers];
+extern THINGINFO_EXTRA gThingInfoExtra[kThingMax];
+extern VECTORINFO_EXTRA gVectorInfoExtra[kVectorMax];
+extern MISSILEINFO_EXTRA gMissileInfoExtra[kMissileMax];
+extern TRPLAYERCTRL gPlayerCtrl[kMaxPlayers];
+extern SPRITEMASS gSpriteMass[kMaxXSprites];
+extern short gProxySpritesList[kMaxSuperXSprites];
+extern short gSightSpritesList[kMaxSuperXSprites];
+extern short gPhysSpritesList[kMaxSuperXSprites];
+extern short gProxySpritesCount;
+extern short gSightSpritesCount;
+extern short gPhysSpritesCount;
+
+// - FUNCTIONS ------------------------------------------------------------------
+bool nnExtEraseModernStuff(spritetype* pSprite, XSPRITE* pXSprite);
+void nnExtInitModernStuff(bool bSaveLoad);
+void nnExtProcessSuperSprites(void);
+bool nnExtIsUmmune(spritetype* pSprite, int dmgType, int minScale = 16);
```

Comment:
Should be Immune?

Reply:
nnExtIsImmune and isImmune acts a bit differently, i left isImmune for Cerberus fix only, and nnExtIsImmune could normally work only ifdefs is enabled

Reply:
I know, but you wrote isUmmune and nnExtIsUmmune instead isImmune and nnExtIsImmune, so just a typo I guess (`I` is next to `U` on the keyboard).

Reply:
oh, thanks :D